PR: https://github.com/odoo/odoo/pull/152517

From: 0ea22b2387d7dafafc1016bb9633077334012bca
From: mathysPaul
Date: 2024-04-02 13:09:23

Breaking data model changes score: 9, change matches:
-    show_credentials_page = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_allow_tokenization = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_allow_express_checkout = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_pre_msg = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_pending_msg = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_auth_msg = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_done_msg = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_cancel_msg = fields.Boolean(compute='_compute_view_configuration_fields')
-    require_currency = fields.Boolean(compute='_compute_view_configuration_fields')

Total Changes: 47

[IMP] payment: Remove view configuration fields

Post-"views-pocalypse", this commit eliminates obsolete view
configuration fields in the payment module. Utilizes the
<attribute add="..." separator="..."/> feature to seamlessly
integrate settings like show_credentials_page, show_allow_tokenization,
and others directly into views via XPath, removing the need for
the _compute_view_configuration_fields method.

task-3679393

closes odoo/odoo#152517

Related: odoo/enterprise#56181
Related: odoo/documentation#7686
Related: odoo/upgrade#5667
Signed-off-by: Antoine Vandevenne (anv) <anv@odoo.com>

================================= pseudo patch: =================================

--- a/addons/payment/models/payment_provider.py
+++ b/addons/payment/models/payment_provider.py
@@ -176,17 +176,6 @@ class PaymentProvider(models.Model):
     module_state = fields.Selection(string="Installation State", related='module_id.state')
     module_to_buy = fields.Boolean(string="Odoo Enterprise Module", related='module_id.to_buy')
 
-    # View configuration fields
-    show_credentials_page = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_allow_tokenization = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_allow_express_checkout = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_pre_msg = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_pending_msg = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_auth_msg = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_done_msg = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_cancel_msg = fields.Boolean(compute='_compute_view_configuration_fields')
-    require_currency = fields.Boolean(compute='_compute_view_configuration_fields')
-
     #=== COMPUTE METHODS ===#
 
     @api.depends('code')
@@ -221,42 +210,6 @@ class PaymentProvider(models.Model):
             elif provider.state == 'enabled':
                 provider.color = 7  # green
 
-    @api.depends('code')
-    def _compute_view_configuration_fields(self):
-        """ Compute the view configuration fields based on the provider.
-
-        View configuration fields are used to hide specific elements (notebook pages, fields, etc.)
-        from the form view of payment providers. These fields are set to `True` by default and are
-        as follows:
-
-        - `show_credentials_page`: Whether the "Credentials" notebook page should be shown.
-        - `show_allow_tokenization`: Whether the `allow_tokenization` field should be shown.
-        - `show_allow_express_checkout`: Whether the `allow_express_checkout` field should be shown.
-        - `show_pre_msg`: Whether the `pre_msg` field should be shown.
-        - `show_pending_msg`: Whether the `pending_msg` field should be shown.
-        - `show_auth_msg`: Whether the `auth_msg` field should be shown.
-        - `show_done_msg`: Whether the `done_msg` field should be shown.
-        - `show_cancel_msg`: Whether the `cancel_msg` field should be shown.
-        - `require_currency`: Whether the `available_currency_ids` field shoud be required.
-
-        For a provider to hide specific elements of the form view, it must override this method and
-        set the related view configuration fields to `False` on the appropriate `payment.provider`
-        records.
-
-        :return: None
-        """
-        self.update({
-            'show_credentials_page': True,
-            'show_allow_tokenization': True,
-            'show_allow_express_checkout': True,
-            'show_pre_msg': True,
-            'show_pending_msg': True,
-            'show_auth_msg': True,
-            'show_done_msg': True,
-            'show_cancel_msg': True,
-            'require_currency': False,
-        })
-
     @api.depends('code')
     def _compute_feature_support_fields(self):
         """ Compute the feature support fields based on the provider.
